modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -0,0 +1,1016 @@
+// to post outputs blobs (cv::GMat's).
+static void PostOutputs(::ov::InferRequest             &infer_request,
+                        std::exception_ptr             eptr,
+                        std::shared_ptr<OVCallContext> ctx) {
+    GAPI_ITT_STATIC_LOCAL_HANDLE(ov_cb_post_outputs_hndl, "OV_async_callback_PostOutputs");
+    GAPI_ITT_AUTO_TRACE_GUARD(ov_cb_post_outputs_hndl);
+
+    ctx->eptr = std::move(eptr);
+    for (auto i : ade::util::iota(ctx->uu.params.num_out)) {
+        // NB: Copy data back only if execution finished sucessfuly.
+        // Otherwise just post outputs to keep streaming executor contract.
+        if (!ctx->eptr) {
+            const auto& out_name = ctx->uu.params.output_names[i];
+            copyFromOV(infer_request.get_tensor(out_name),
+                       ctx->outMatR(i));
+        }
+        auto output = ctx->output(i);
+        ctx->out.meta(output, ctx->getMeta());
+        ctx->out.post(std::move(output), ctx->eptr);
+    }
+}
+
+namespace cv {
+namespace gimpl {
+namespace ov {
+
+template <typename Attr>
+using AttrMap = cv::gapi::ov::detail::AttrMap<Attr>;
+
+template <typename Attr>
+using LayerVariantAttr = cv::gapi::ov::detail::LayerVariantAttr<Attr>;
+
+template <typename Attr> AttrMap<Attr>
+broadcastLayerAttr(const LayerVariantAttr<Attr>   &layer_attr,
+                   const std::vector<std::string> &layer_names) {
+    AttrMap<Attr> map;
+    if (cv::util::holds_alternative<AttrMap<Attr>>(layer_attr)) {
+        map = cv::util::get<AttrMap<Attr>>(layer_attr);
+        // NB: Validate map:
+        std::unordered_set<std::string> existing_layers =
+            {layer_names.begin(), layer_names.end()};
+
+        for (const auto &p : map) {
+            const auto it = existing_layers.find(p.first);
+            if (it == existing_layers.end()) {
+                cv::util::throw_error(
+                        std::logic_error("OV Backend: Failed to"
+                                         " find layer with name: " + p.first));
+            }
+        }
+    } else if (cv::util::holds_alternative<Attr>(layer_attr)) {
+        // NB: Broadcast value to all layers.
+        auto elem = cv::util::get<Attr>(layer_attr);
+        for (auto &&layer_name : layer_names) {
+            map.emplace(layer_name, elem);
+        }
+    }
+    return map;
+}
+
+template <typename K, typename V>
+cv::optional<V> lookUp(const std::map<K, V> &map, const K& key) {
+    const auto it = map.find(key);
+    if (it == map.end()) {
+        return {};
+    }
+    return cv::util::make_optional(std::move(it->second));
+}
+
+static bool isImage(const cv::GMatDesc &desc,
+                    const ::ov::Shape  &model_shape) {
+    return (model_shape.size() == 4u)                      &&
+           (!desc.isND())  /* dims == 2 */                 &&
+           (desc.chan == 1 || desc.chan == 3)              &&
+           (desc.size.height != 1 && desc.size.width != 1) &&
```

Comment:
Btw do you remember if we might have any non-1 dims in some of our cases which are still tensors? (e.g. 2, 3, other small numbers which are definitely are not images)

Reply:
We can have `Mat({15, 14}, CV_8UC1)` but in this case model apparently will have `2D` dimensions. so that it would be `TENSOR` for us